genome_designer/genome_finish/assembly.py

```diff
@@ -1,169 +1,379 @@
+    }
+}
 
+NUM_CONTIGS_TO_EVALUATE = 1000
 
-def generate_contigs(experiment_sample_to_alignment, contig_label_base):
 
-    timestamp = str(datetime.datetime.now())
+def kmer_coverage(C, L, k):
+    """Converts contig coverage to kmer coverage
+
+    Args:
+        * C: Coverage
+        * L: Template length
+        * k: hash length
+    """
+    return C * (L - k + 1) / float(L)
+
+
+def generate_contigs(sample_alignment,
+        sv_read_classes={}, input_velvet_opts={},
+        overwrite=False):
+
+    contig_label_base = '' # Force for now
 
-    # Grab reference genome fasta path
-    reference_genome = (
-            experiment_sample_to_alignment.alignment_group.reference_genome)
+    # Grab reference genome fasta path, ensure indexed
+    reference_genome = sample_alignment.alignment_group.reference_genome
     ref_fasta_dataset = reference_genome.dataset_set.get_or_create(
             type=Dataset.TYPE.REFERENCE_GENOME_FASTA)[0]
     prepare_ref_genome_related_datasets(reference_genome, ref_fasta_dataset)
 
     # Make data_dir directory to house genome_finishing files
-    genome_finishing_dir = os.path.join(
-            experiment_sample_to_alignment.get_model_data_dir(),
-            'genome_finishing')
+    assembly_dir = os.path.join(
+            sample_alignment.get_model_data_dir(),
+            'assembly')
 
-    # Make data_dir directory if it does not exist
-    if not os.path.exists(genome_finishing_dir):
-        os.mkdir(genome_finishing_dir)
+    # Make assembly directory if it does not exist
+    if not os.path.exists(assembly_dir):
+        os.mkdir(assembly_dir)
 
-    data_dir = os.path.join(genome_finishing_dir, '0')
     data_dir_counter = 0
+    data_dir = os.path.join(assembly_dir, str(data_dir_counter))
     while(os.path.exists(data_dir)):
         data_dir_counter += 1
-        data_dir = os.path.join(genome_finishing_dir, str(data_dir_counter))
+        data_dir = os.path.join(assembly_dir, str(data_dir_counter))
     os.mkdir(data_dir)
 
-    # Retrieve bwa mem .bam alignment if exists otherwise generate it
-    if not experiment_sample_to_alignment.dataset_set.filter(
-            type=Dataset.TYPE.BWA_ALIGN).exists():
-        add_dataset_to_entity(
-                experiment_sample_to_alignment,
-                'sample_alignment_for_assembly',
-                Dataset.TYPE.BWA_ALIGN)
-        align_with_bwa_mem(
-                experiment_sample_to_alignment.alignment_group,
-                experiment_sample_to_alignment,
-                project=reference_genome.project)
+    # Get a bam of sorted SV indicants with pairs
+    sv_indicants_bam = get_sv_indicating_reads(sample_alignment,
+            sv_read_classes, overwrite=overwrite)
 
-    # Grab alignment bam file-path
-    alignment_sorted = get_dataset_with_type(
-                experiment_sample_to_alignment,
-                Dataset.TYPE.BWA_ALIGN).get_absolute_location()
+    velvet_opts = DEFAULT_VELVET_OPTS
```

Comment:
Make a copy of the default dictionary so you don't modify it. Looks like there's no bug here, but it's good practice to avoid accidentally mutating something you didn't mean to.

    velvet_opts = dict(DEFAULT_VELVET_OPTS)